### DISCUSSION TITLE
ScriptReference/ForceMode

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#320] [Notes]
# ScriptReference/ForceMode

https://docs.unity3d.com/ScriptReference/ForceMode.html

**Comment by docscomments** [comment]
**Comment by `@bgr`:**  

[Examples of using all modes to achieve the same result.](https://twitter.com/ykravchik/status/851721192415383552)

**Comment by vertxxyz** [comment]
The classic forcemode matrix by [nothke](https://twitter.com/nothke):
|   | Time dependent  | Time independent  |   |
|---|---|---|---|
|  **Mass dependent** |  **Force**<br>Same as:<br>`v += f * dt / m` |  **Impulse**<br>Same as:<br>`v += f / m` |  When force depends on mass.<br>The same amount of force moves small objects faster, and big ones slower.<br>For example, different projectiles used with the same amount of gunpowder. |
|  **Mass independent** |  **Acceleration**<br>Same as:<br>`v += f * dt` | **VelocityChange**<br>Same as:<br>`v += f`  |  When the mass doesn't affect the change of velocity.<br>Gravity, for example, accelerates all objects equally. |
|   | When force is applied continuously, each frame.<br>A rocket engine, jetpack, magnet... | When force is applied instantaneously.<br>A response to a collision, shooting a bullet, kicking a ball...  |   |
